Dockerfile.validate

```diff
@@ -1,5 +1,6 @@
 FROM quay.io/app-sre/golang:1.18.7 as validator
 WORKDIR /build
 COPY . .
-RUN make test-schema
-
+RUN go install github.com/golang/mock/mockgen && \
+    go get github.com/Khan/genqlient && \
```

Comment:
If we just need to install some executables, then `go install` should be used, [doc](https://go.dev/doc/go-get-install-deprecation).

`go install` should also specify a version.

Also should move `go install` stage before `COPY` to take advantage of docker build cache.

So the final Dockerfile can be

```dockerfile
FROM quay.io/app-sre/golang:1.18.7 as validator
WORKDIR /build
RUN go install github.com/golang/mock/mockgen@v1.6.0 && \
    go install github.com/Khan/genqlient@v0.5.0
COPY . .
RUN make generate gotest
```

Another improvement is to merge this `Dockerfile.validate` with `Dockerfile` to use multi-stage build to remove duplicates.

Reply:
Thanks! I will read up on `go install` and build in the other changes

Reply:
`COPY . .` will almost always invalidate cache because there is no `.dockerignore` file, I can address it in a seperate PR once this one is merged.